test/dhcp/subnet_service_test.rb

```diff
@@ -49,6 +49,59 @@ def test_find_subnet
     assert_equal subnets.first, @service.find_subnet("192.168.0.0")
   end
 
+  def test_find_subnet_with_cidr_mask
+    subnets = [Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.64", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.128", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.192", "255.255.255.192")]
+    @service.add_subnets(*subnets)
+    assert_equal subnets[2], @service.find_subnet("192.168.0.131")
+  end
+
+  def test_find_subnet_with_cidr_mask_and_match_in_a_group_of_one_network
```

Comment:
This test is probably not needed -- I originally added it to test binary-search for when a partition contained a single network only.